tests/unit/pip_sync_faster/main_test.py

```diff
@@ -1,6 +1,156 @@
+
+    def test_if_theres_one_missing_hash(
+        self, assert_hashes_cached, cache_hashes, requirements_files, run
+    ):
+        # Make the cached hashes file be missing one hash.
+        cache_hashes(main.get_hashes(requirements_files[1:]))
+
+        main.pip_sync_maybe(requirements_files)
+
+        run.assert_called_once_with(["pip-sync", *requirements_files], check=True)
+        assert_hashes_cached(requirements_files)
+
+    def test_if_theres_a_different_files_hash_cached(
+        self, assert_hashes_cached, cache_hashes, requirements_files, run
+    ):
+        # The cache contains correct hashes for both dev.txt and format.txt.
+        cache_hashes(main.get_hashes(requirements_files))
+
+        # pip-sync-faster is called with just format.txt.
+        main.pip_sync_maybe(requirements_files[1:])
+
+        # It should call pip-sync: the last time that pip-sync-faster was
+        # called (and updated the cache) it was called with both dev.txt and
+        # format.txt. Now it's being called with just format.txt. Any
+        # requirements that're in dev.txt but not in format.txt need to be
+        # removed from the venv.
+        run.assert_called_once_with(["pip-sync", *requirements_files[1:]], check=True)
+        # It should update the cache to contain only format.txt.
+        assert_hashes_cached(requirements_files[1:])
+
+    @pytest.fixture
+    def cached_hashes_path(self, tmp_path):
+        """Return the path where pip-sync-faster will look for its cached hashes file."""
+        return tmp_path / "pip_sync_faster.json"
+
+    @pytest.fixture
+    def assert_hashes_cached(self, cached_hashes_path):
+        def assert_hashes_cached(requirements_files):
+            """Assert the cache contains correct hashes for the given requirements files."""
+            with open(cached_hashes_path, "r", encoding="utf-8") as handle:
+                cached_hashes = json.load(handle)
+            assert cached_hashes == main.get_hashes(requirements_files)
+
+        return assert_hashes_cached
+
+    @pytest.fixture
+    def cache_hashes(self, cached_hashes_path):
+        def cache_hashes(hashes):
+            """Cache the given hashes for the given paths."""
+            with open(cached_hashes_path, "w", encoding="utf-8") as handle:
+                json.dump(hashes, handle)
+
+        return cache_hashes
+
+    @pytest.fixture
+    def requirements_files(self, tmp_path):
+        """Create the test requirements files and return their paths."""
+        dev_txt_path = tmp_path / "dev.txt"
+        with open(dev_txt_path, "w", encoding="utf8") as handle:
+            handle.write("This is a fake dev.txt requirements file.")
+
+        format_txt_path = tmp_path / "format.txt"
+        with open(format_txt_path, "w", encoding="utf8") as handle:
+            handle.write("This is a fake format.txt requirements file.")
+
+        return [dev_txt_path, format_txt_path]
+
+
+class TestEntryPoint:
```

Comment:
In many cases I think this is something you _could_ skip, but I think testing it in this case is actually a good idea as we're doing slightly fancy arg merging stuff.